### DISCUSSION TITLE
Migrating an GraphQL-QML Python framework to this project.

### DISCUSSION THREAD

**Discussion by nrbnlulu** [woboq/qmetaobject-rs#282] [General]
I am writing [micro framework for GraphQL-QML  in Python ](https://github.com/nrbnlulu/qtgql)when it gets mature I intend to port it to cpp/rust.

Though if I use this project I have few considerations:
1. I need to maintain a store of objects (nodes if you are familiar with react-relay) by their ID
and from what I know rust has no concept of pointers so I would have to query the store every time 
I need to use data from a node am I right? wouldn't that impose performance issues? 

2. What alternatives do you have for I/O / network operations since this project (from what i'v seen) doesn't have bindings for 
other Qt's API apart from QML.
  
  _sidenote: I have very small experience with C++ and no experience with rust._

**Comment by ogoffart** [comment]
> from what I know rust has no concept of pointers

Rust has the concept of pointer. But also references,  `Rc<>`, `Arc<>`, hash table, index in vectors. So you're covered.

> What alternatives do you have for I/O / network operations since this project (from what i'v seen) doesn't have bindings for
other Qt's API apart from QML.

The whole set of libraries on https://crates.io

**Reply by nrbnlulu** [reply]
> The whole set of libraries on https://crates.io/

Yeah though you would have to use sync only crates no?

**Reply by ogoffart** [reply]
I don't know what you mean by "sync only crates"

The point is that this crate helps for QML to build an UI in pure rust. The Rust ecosystem already have enough libraries to do most of what other part of Qt does.

**Reply by nrbnlulu** [reply]
I mean that using rust for network operation might block Qt's eventloop no?
